wiki/Installing-Oppia-Android.md

```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
 
 ## Set up and run tests
 Testing the app is an integral part of our development process. You will need to test all code changes to ensure that the app works correctly, therefore it is important to ensure that your test configuration works.
 
-We strongly recommend running tests on Robolectric which is faster because it does not require a physical device or emulator setup.
+Our Bazel setup currently supports running tests on Robolectric which is fast because it does not require a physical device or emulator setup.
 
-### Configure Robolectric Tests
+We can run tests either through the UI supported by the Bazel Plugin, or via running a `bazel test` command on the terminal, i.e.:
 
-#### Step-by-Step guidance for setting up and running app modules robolectric test:
+> Tip 1: In order to get the correct path always, open the test file, then on the AS sidebar, right click on the file, and select `copy path/reference`. Then select `copy path from repository root`. Paste the copied path to terminal and remove the `.kt` extension.
 
-1. Go to **Edit Configuration** in Android Studio (Bumblebee | 2021.1.1 Patch 3)
-   ![](https://user-images.githubusercontent.com/9396084/79109714-83525980-7d96-11ea-99d7-f83ea81a8a50.png)
+   ```
+   bazel test full-path-of-test-file
+   ```
 
-2. Click on Add(+) -> **JUnit**
-   ![](https://github.com/oppia/oppia-android/assets/76530270/87caf3fc-37d9-472d-92fd-b8ec49fb6b49)
+e.g.
+   
+   ```
+   bazel test domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest
+   ```
 
-3. Enter following information:
-   - a) Name of test. Example: In my case "SplashActivityTest"
-   - b) Make sure select "java 11" and oppia-android.app
-   - c) Class path of Test class. Example: In my case "org.oppia.android.app.splash.
-   SplashActivityTest"
-   - d) Press `OK` to select the test.
-   ![](https://github.com/oppia/oppia-android/assets/76530270/5901624a-df76-4b27-8f31-6077a68fcb89)
+> Tip 2: You can pass a target argument to the `bazel test` command instead. Copy the relative path to the test file and query its target using `bazel query path-of-test-file` e.g. `bazel query domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt`. This will give you a Bazel target (i.e. starts with '//') --just copy that and remove the `.kt` at the end and you can pass it to `bazel test`, e.g.
 
-4. Click on "Run" button to run robolectric test. (In my case "SplashActivityTest")
-   ![](https://github.com/oppia/oppia-android/assets/76530270/75a6b998-90c5-4f0a-8886-78f96970be90)
+```
+bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest
+```
 
-#### For tests that are in non-app modules, such as **domain** or **utility**::
+### Configure Robolectric Tests
```

Comment:
Probably also better to put in the user guide?

Reply:
The whole setting up tests section?

Reply:
I kept the basic test instructions here and moved the rest of the instructions and tips to the user guide.